mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -230,6 +231,44 @@ private void resolve(ComponentModel model, Object owner) {
     return result.get();
   }
 
+  /**
+   * @param extensionModel the model which owns the {@code component}
+   * @param component a component
+   * @return The config (if any) that the {@code component} needs in order to function
+   */
+  public static Optional<ConfigurationModel> getConfigurationFromComponent(ExtensionModel extensionModel, NamedObject component) {
```

Comment:
this method is fundamentally wrong. The same component can be a part of several configurations. An what you're getting is configurations for a component, not from it.